examples/http_auth_random.rs

```diff
@@ -20,10 +20,23 @@ use std::time::Duration;
 #[no_mangle]
 pub fn _start() {
     proxy_wasm::set_log_level(LogLevel::Trace);
-    proxy_wasm::set_http_context(|_, _| -> Box<dyn HttpContext> { Box::new(HttpAuthRandom) });
+    proxy_wasm::set_root_context(|_| -> Box<dyn RootContext> { Box::new(HttpAuthRandomRoot) });
```

Comment:
Looking at this example, I wonder if we couldn't do something type-specific here, e.g.
```
proxy_wasm::set_root_context(|_| -> Box<dyn RootContext> { Box::new(HttpHeadersRoot) });
proxy_wasm::set_http_context(|root_context, context_id| -> Box<dyn HttpContext> {
    root_context.new_http_context(context_id)
});
```
with dispatcher taking care of resolving `root_context_id` to `root_context` before calling this callback?

Reply:
On the second thought, that might be useless, since we cannot expose multiple interfaces right now anyway.